adapter/policy/handler/utils.go

```diff
@@ -26,6 +29,14 @@ func parseTarget(target []v1.TargetElement, namespace string) []policy.Endpoint
 	return endpoints
 }
 
+func GetKubeSecret(kubeClient kubernetes.Interface, namespace string, ref v1.ClientSecretRef) (*k8sv1.Secret, error) {
+	secret, err := kubeClient.CoreV1().Secrets(namespace).Get(ref.Name, metav1.GetOptions{})
```

Comment:
looks to me like you can just return the call.

```suggestion
	return kubeClient.CoreV1().Secrets(namespace).Get(ref.Name, metav1.GetOptions{})
```